### PR TITLE
new package: es

### PR DESCRIPTION
**Pull request by sjmackenzie**

### CODE REVIEW

File: pkgs/shells/essh/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    url = "git://github.com/wryun/es-shell";
+    rev = "fdf29d5296ce3a0ef96d2b5952cff07878753975";
+    sha256 = "1hj0g8r59ry9l50h4gdal38nf8lvb3cgl6c9bx5aabkw5i778dfk";
+  };
+
+  preConfigure =
+    ''
+      aclocal
+      autoconf
+      libtoolize -qi
```
Comment (domenkozar) [inline pkgs/shells/essh/default.nix]: Same can be done by passing `autoreconfHook` to `buildInputs`

File: pkgs/shells/essh/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      mkdir -p $out/bin
+      mkdir -p $out/man
```
Comment (domenkozar) [inline pkgs/shells/essh/default.nix]: `mkdir -p $out/{bin,man}` is really an installation step

File: pkgs/shells/essh/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      configureFlags="--with-readline --prefix=$out --bindir=$out/bin --mandir=$out/man"
```
Comment (domenkozar) [inline pkgs/shells/essh/default.nix]: `configureFlags` can be passed directly to `mkDerivation`

File: pkgs/shells/essh/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchgit, readline, autoconf, automake, libtool, yacc }:
+
+let
+  version = "fdf29d";
```
Comment (domenkozar) [inline pkgs/shells/essh/default.nix]: we prefer dates over git versions, not sure where the docs really are for that though.

File: pkgs/shells/essh/default.nix (hunk 3 above)
Comment (sjmackenzie) [inline pkgs/shells/essh/default.nix]: that fails with: 
```
./esdump < ./initial.es > initial.c
/bin/pwd: No such file or directory
/bin/date: No such file or directory
gcc  -I. -I. -W -Wall -Wno-missing-field-initializers -Wno-unused-parameter -Wno-clobbered -g -O2   -c -o initial.o initial.c
gcc -o es   access.o closure.o conv.o dict.o eval.o except.o fd.o gc.o glob.o glom.o input.o heredoc.o list.o main.o match.o open.o opt.o prim-ctl.o prim-etc.o prim-io.o prim-sys.o prim.o print.o proc.o sigmsgs.o signal.o split.o status.o str.o syntax.o term.o token.o tree.o util.o var.o vec.o version.o y.tab.o initial.o  -lreadline 
installing
install flags: install SHELL=/nix/store/kz4aii9y5kdxilxrjdz55gdr0piziwcz-bash-4.3-p33/bin/bash    
/nix/store/ym4xn5j5nkxrcg5wfmdkfnj4jrfs16f1-coreutils-8.23/bin/install -c -s ./es ut/bin
/nix/store/ym4xn5j5nkxrcg5wfmdkfnj4jrfs16f1-coreutils-8.23/bin/install: cannot create regular file 'ut/bin': No such file or directory
Makefile:85: recipe for target 'install' failed
make: *** [install] Error 1
builder for ‘/nix/store/5brsnflcfs9xjr9ffdrarmbaq1q5jgsp-essh-git-2015-04-11.drv’ failed with exit code 2
error: build of ‘/nix/store/5brsnflcfs9xjr9ffdrarmbaq1q5jgsp-essh-git-2015-04-11.drv’ failed
```

File: pkgs/shells/essh/default.nix (hunk 2 above)
Comment (sjmackenzie) [inline pkgs/shells/essh/default.nix]: done

File: pkgs/shells/essh/default.nix (hunk 1 above)
Comment (sjmackenzie) [inline pkgs/shells/essh/default.nix]: okay that falls over with 
```
autoreconfPhase
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /nix/store/njr9a876l3p13ylj0vs8kbxc4lq6v5jj-autoconf-2.69/bin/autoconf --force
autoreconf: running: /nix/store/njr9a876l3p13ylj0vs8kbxc4lq6v5jj-autoconf-2.69/bin/autoheader --force
autoheader: warning: missing template: ABUSED_GETENV
autoheader: Use AC_DEFINE([ABUSED_GETENV], [], [Description])
autoheader: warning: missing template: HAVE_DEV_FD
autoheader: warning: missing template: KERNEL_POUNDBANG
autoheader: warning: missing template: LIMIT_T
autoheader: warning: missing template: NO_VA_LIST_ASSIGN
autoheader: warning: missing template: SOLARIS
autoheader: warning: missing template: VOID_SIGNALS
autoreconf: /nix/store/njr9a876l3p13ylj0vs8kbxc4lq6v5jj-autoconf-2.69/bin/autoheader failed with exit status: 1
builder for ‘/nix/store/sbihz82v5h8wqw6nncdw4s2fwigmjlc6-essh-git-2015-04-11.drv’ failed with exit code 1
error: build of ‘/nix/store/sbihz82v5h8wqw6nncdw4s2fwigmjlc6-essh-git-2015-04-11.drv’ failed
```

File: pkgs/shells/essh/default.nix (hunk 4 above)
Comment (sjmackenzie) [inline pkgs/shells/essh/default.nix]: https://github.com/NixOS/nixpkgs/pull/6539

File: pkgs/shells/essh/default.nix (hunk 1 above)
Comment (domenkozar) [inline pkgs/shells/essh/default.nix]: Huh, no idea.

File: pkgs/shells/essh/default.nix (hunk 3 above)
Comment (domenkozar) [inline pkgs/shells/essh/default.nix]: You'll probably need to do `--mandir=$(out)/man`

File: pkgs/shells/essh/default.nix (hunk 1 above)
Comment (sjmackenzie) [inline pkgs/shells/essh/default.nix]: yeah... I'm not even going to mess around with it... it 4am my side :-)

File: pkgs/shells/essh/default.nix (hunk 3 above)
Comment (sjmackenzie) [inline pkgs/shells/essh/default.nix]: okay that did it.

**Comment by sjmackenzie** [comment]
@pSub @7c6f434c @domenkozar

**Pull request merged**